app/src/main/java/org/wikipedia/random/RandomFragment.kt

```diff
@@ -0,0 +1,301 @@
+package org.wikipedia.random
+
+import android.graphics.drawable.Animatable
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.app.ActivityOptionsCompat
+import androidx.core.os.bundleOf
+import androidx.core.util.Pair
+import androidx.core.view.ViewCompat
+import androidx.fragment.app.Fragment
+import androidx.viewpager2.widget.ViewPager2.OnPageChangeCallback
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
+import io.reactivex.rxjava3.core.Observable
+import io.reactivex.rxjava3.disposables.CompositeDisposable
+import io.reactivex.rxjava3.functions.Consumer
+import io.reactivex.rxjava3.schedulers.Schedulers
+import org.wikipedia.Constants
+import org.wikipedia.Constants.InvokeSource
+import org.wikipedia.R
+import org.wikipedia.WikipediaApp
+import org.wikipedia.analytics.RandomizerFunnel
+import org.wikipedia.databinding.FragmentRandomBinding
+import org.wikipedia.dataclient.WikiSite
+import org.wikipedia.events.ArticleSavedOrDeletedEvent
+import org.wikipedia.history.HistoryEntry
+import org.wikipedia.page.ExclusiveBottomSheetPresenter
+import org.wikipedia.page.PageActivity
+import org.wikipedia.page.PageTitle
+import org.wikipedia.readinglist.AddToReadingListDialog
+import org.wikipedia.readinglist.LongPressMenu
+import org.wikipedia.readinglist.MoveToReadingListDialog
+import org.wikipedia.readinglist.ReadingListBehaviorsUtil
+import org.wikipedia.readinglist.ReadingListBehaviorsUtil.AddToDefaultListCallback
+import org.wikipedia.readinglist.ReadingListBehaviorsUtil.addToDefaultList
+import org.wikipedia.readinglist.database.ReadingListDbHelper
+import org.wikipedia.readinglist.database.ReadingListPage
+import org.wikipedia.util.AnimationUtil.PagerTransformer
+import org.wikipedia.util.DimenUtil
+import org.wikipedia.util.FeedbackUtil
+import org.wikipedia.util.log.L
+import org.wikipedia.views.PositionAwareFragmentStateAdapter
+
+class RandomFragment : Fragment() {
+
+    companion object {
+        fun newInstance(wikiSite: WikiSite, invokeSource: InvokeSource) = RandomFragment().apply {
+            arguments = bundleOf(
+                    RandomActivity.INTENT_EXTRA_WIKISITE to wikiSite,
+                    Constants.INTENT_EXTRA_INVOKE_SOURCE to invokeSource
+            )
+        }
+    }
+
+    private var _binding: FragmentRandomBinding? = null
+    private val binding get() = _binding!!
+
+    private val disposables = CompositeDisposable()
+
+    private var wikiSite: WikiSite? = null
+    private val requireWikiSite get() = wikiSite ?: throw IllegalStateException("wikiSite is null.")
```

Comment:
I'm not sure we need this kind of construct. Let's just make `wikiSite` itself into a `lateinit var`, and let it be non-null.

Reply:
There are two options:

1. `wikiSite = requireArguments().getParcelable(RandomActivity.INTENT_EXTRA_WIKISITE)!!`

2.
```
wikiSite = requireArguments().getParcelable(RandomActivity.INTENT_EXTRA_WIKISITE) 
                ?: throw IllegalStateException("wikiSite is null.")
```

I personally like the second more because it gives additional information.

Reply:
I personally prefer the first one, since it will give us nearly the same information.